batch7crud-roll8/src/main/java/com/lftechnology/batch7crud/dao/EmployeeDao.java

```diff
@@ -0,0 +1,182 @@
+        employee.setId(resultSet.getInt(1));
+      }
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+    finally {
+      try {
+        if (resultSet != null)
+          resultSet.close();
+      } catch (Exception e) {
+        LOGGER.log(Level.SEVERE, e.getMessage(), e);
+        throw new DataException(e.getMessage());
+      }
+    }
+  }
+
+  public List<Employee> fetch(Integer pageLimit, Integer offset) throws DataException {
+    List<Employee> employeeList = new ArrayList<>();
+    ResultSet resultSet = null;
+
+    try (Connection conn = DBConnection.getConnection();
+        PreparedStatement statement = conn.prepareStatement(EMPLOYEE_SELECT_ALL)
+    ){
+      statement.setInt(1, pageLimit);
+      statement.setInt(2, offset);
+      resultSet = statement.executeQuery();
+      while (resultSet.next()) {
+        Employee employee = new Employee();
+        employee.setId(resultSet.getInt(ModelConstants.ID));
+        employee.setName(resultSet.getString(ModelConstants.NAME));
+        employee.setAddress(resultSet.getString(ModelConstants.ADDRESS));
+        employee.setDesignation(resultSet.getString(ModelConstants.DESIGNATION));
+        employee.setPhone(resultSet.getString(ModelConstants.PHONE));
+        employeeList.add(employee);
+      }
+      return employeeList;
+    } catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+    finally {
+      try {
+        if (resultSet != null)
+          resultSet.close();
+      } catch (Exception e) {
+        LOGGER.log(Level.SEVERE, e.getMessage(), e);
+        throw new DataException(e.getMessage());
+      }
+    }
+  }
+
+  public Employee fetchById(Integer id) throws DataException {
+    Employee employee = null;
+    ResultSet resultSet = null;
+
+    try (Connection conn = DBConnection.getConnection();
+      PreparedStatement statement = conn.prepareStatement(EMPLOYEE_SELECT_BY_ID)
+    ){
+      statement.setInt(1, id);
+      resultSet = statement.executeQuery();
+      if (resultSet.next()) {
+        employee = new Employee();
+        employee.setId(resultSet.getInt(ModelConstants.ID));
+        employee.setName(resultSet.getString(ModelConstants.NAME));
+        employee.setAddress(resultSet.getString(ModelConstants.ADDRESS));
+        employee.setDesignation(resultSet.getString(ModelConstants.DESIGNATION));
+        employee.setPhone(resultSet.getString(ModelConstants.PHONE));
+      }
+      return employee;
+    }
+    catch (SQLException e) {
+      LOGGER.log(Level.SEVERE, e.getMessage(), e);
+      throw new DataException(e.getMessage());
+    }
+    finally {
+      try {
+        if (resultSet != null)
+          resultSet.close();
+      } catch (Exception e) {
+        LOGGER.log(Level.SEVERE, e.getMessage(), e);
+        throw new DataException(e.getMessage());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not throw exceptions in finally blocks. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1163)